### ISSUE TITLE
Corner cases of xfab.tools.u_to_euler()

### ISSUE THREAD

**Issue by AxelHenningsson** [FABLE-3DXRD/xfab#12]
Hey, great libs here all around. Appreciate!

Small improvement to make in xfab.tools.u_to_euler: Think it is broken for some corner cases like the minimal example below

```
import numpy as np
from xfab import tools

phi1, PHI, phi2 = np.radians([190.,0, 0])
U = tools.euler_to_u( phi1,PHI,phi2 )
phi1_new, PHI_new, phi2_new = tools.u_to_euler( U )
U_new = tools.euler_to_u( phi1_new, PHI_new, phi2_new )

print('Original Euler angles',phi1, PHI, phi2)
print('New Euler angles',phi1_new, PHI_new, phi2_new)
print(' ')
print('Original orientation matrix:')
print(U)
print('New orientation matrix')
print(U_new)
```

```
Original Euler angles 3.3161255787892263 0.0 0.0
New Euler angles 0.0 0.0 0.0
 
Original orientation matrix:
[[-0.98480775  0.17364818 -0.        ]
 [-0.17364818 -0.98480775  0.        ]
 [ 0.          0.          1.        ]]
New orientation matrix
[[ 1. -0.  0.]
 [ 0.  1. -0.]
 [ 0.  0.  1.]]
```
Easy fix is probably to add a return in the first if in the method? Seems these are the correct phi1 and phi2 in these cases and no further computations need be done?

Cheers
Axel Henningsson

**Comment by jonwright** [comment]
Hi Axel,
Thanks! Can you send a pull request with a testcase? Like that you get the credit for fixing it too...!

**Comment by AxelHenningsson** [comment]
Hi Jon,

Pull request created! Turns out this was a bit more complicated than at first glance. These notes where very helpful:
[Depriester, Dorian. (2018). Computing Euler angles with Bunge convention from rotation matrix](https://www.researchgate.net/publication/324088567_Computing_Euler_angles_with_Bunge_convention_from_rotation_matrix)

if you don't want the Euler angles in the range 0 to 2*pi, the implementation could be slightly changed. I think this one is nice though, obviously.

Cheers
Axel

**Comment by jonwright** [comment]
So I think this was fixed in #13
